### PR TITLE
Remove duplicated Numa detection code.

### LINKED ISSUE #29968: Issue with adding additional CFLAGS path
I would like to know why adding additional include paths via CFLAGS doesn't work for building Pytorch?
I haven't root access to install devel packages, so I have the downloaded rpm file and extract it inmy home.
```
$ CFLAGS="-I/storage/users/mnaderan/tools/usr/include/" python setup.py install
$ ls /storage/users/mnaderan/tools/usr/include/
numacompat1.h  numa.h  numaif.h
```
Still I get `numa.h not found`

```
-- Performing Test HAVE_POSIX_REGEX
-- Performing Test HAVE_POSIX_REGEX
-- Performing Test HAVE_POSIX_REGEX -- compiled but failed to run
CMake Error at third_party/benchmark/CMakeLists.txt:231 (message):
  Failed to determine the source files for the regular expression backend


-- Configuring incomplete, errors occurred!
See also "/storage/users/mnaderan/pt/pytorch/build/CMakeFiles/CMakeOutput.log".
See also "/storage/users/mnaderan/pt/pytorch/build/CMakeFiles/CMakeError.log".
Traceback (most recent call last):
  File "setup.py", line 751, in <module>
    build_deps()
  File "setup.py", line 310, in build_deps
    cmake=cmake)
  File "/storage/users/mnaderan/pt/pytorch/tools/build_pytorch_libs.py", line 56, in build_caffe2
    rerun_cmake)
  File "/storage/users/mnaderan/pt/pytorch/tools/setup_helpers/cmake.py", line 318, in generate
    self.run(args, env=my_env)
  File "/storage/users/mnaderan/pt/pytorch/tools/setup_helpers/cmake.py", line 142, in run
    check_call(command, cwd=self.build_dir, env=env)
  File "/usr/lib64/python2.7/subprocess.py", line 542, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '-DBUILD_PYTHON=True', '-DBUILD_TEST=True', '-DCMAKE_BUILD_TYPE=Release', '-DCMAKE_INSTALL_PREFIX=/storage/users/mnaderan/pt/pytorch/torch', '-DCMAKE_PREFIX_PATH=/usr/lib/python2.7/site-packages', '-DNUMPY_INCLUDE_DIR=/storage/users/mnaderan/.local/lib/python2.7/site-packages/numpy/core/include', '-DPYTHON_EXECUTABLE=/bin/python', '-DPYTHON_INCLUDE_DIR=/usr/include/python2.7', '-DPYTHON_LIBRARY=/usr/lib64/libpython2.7.so.1.0', '-DTORCH_BUILD_VERSION=1.4.0a0+f2a35db', '-DUSE_CUDA=True', '-DUSE_NUMPY=True', '/storage/users/mnaderan/pt/pytorch']' returned non-zero exit status 1
```

### PR DESCRIPTION
**Pull request by xuhdev**
cmake/Dependencies.cmake (https://github.com/pytorch/pytorch/blob/1111a6b8100386af82f5612ef551004188cf396c/cmake/Dependencies.cmake#L595-L609) has already detected Numa. Duplicated detection and variables may lead to
incorrect results.

Close #29968

### CODE REVIEW

**Comment by jerryzh168** [comment]
Is there any code that use `CAFFE2_DISABLE_NUMA` to disable numa?

File: c10/macros/cmake_macros.h.in [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #cmakedefine C10_BUILD_SHARED_LIBS
 #cmakedefine C10_USE_GLOG
 #cmakedefine C10_USE_GFLAGS
-#cmakedefine C10_DISABLE_NUMA
+#cmakedefine USE_NUMA @USE_NUMA@
```
Comment (ezyang) [inline c10/macros/cmake_macros.h.in]: So, if I understand correctly, you renamed the macro here so you wouldn't have to rename all the `USE_NUMA` sites?

Comment (xuhdev) [reply c10/macros/cmake_macros.h.in]: (I've corrected this line) There is one place that uses `C10_DISABLE_NUMA` and I have changed that accordingly.

**Comment by xuhdev** [comment]
> Is there any code/script that use `CAFFE2_DISABLE_NUMA` to disable numa?

No, AFAIK, the deleted portion is the only place that uses this variable. It has only been used to set `C10_DISABLE_NUMA`, which is in turn only used in one spot in numa.cpp.

**Comment by ezyang** [comment]
> No, AFAIK, the deleted portion is the only place that uses this variable. It has only been used to set C10_DISABLE_NUMA, which is in turn only used in one spot in numa.cpp.

Err, to put it differently, why can't we keep the version of the macro that is prefixed with C10, for better portability?

**Comment by jerryzh168** [comment]
I mean what if other teams used this flag to disable numa, is there a way we can check?

**Comment by xuhdev** [comment]
> Err, to put it differently, why can't we keep the version of the macro that is prefixed with C10, for better portability?

Sure we can. I've updated the PR

**Comment by xuhdev** [comment]
> I mean what if other teams used this flag to disable numa, is there a way we can check?

`#if defined(C10_USE_NUMA)` should do the trick

**Comment by xuhdev** [comment]
@jerryzh168 OK I see the confusion. Prior to this PR, (a) Dependencies.cmake detects Numa, and modifies `USE_NUMA` accordingly, and (b) MiscCheck.cmake also detects Numa, in a way that is less reliable and inconsistent with the check in Dependencies.cmake, and stores the detection results in `CAFFE2_DISABLE_NUMA`.

This PR removes (b) and keeps (a). `USE_NUMA` should be expected to store the result of the detection, which is in turn transferred to `C10_USE_NUMA`.

**Review by ezyang — APPROVED** [review]

**Comment by xuhdev** [comment]
Rebased to resolve a conflict

**Comment by xuhdev** [comment]
Resolved a conflict

**Comment by ezyang** [comment]
Sorry this one fell of the land wagon. I won't be able to land it until the NY (in China) right now, maybe find someone else to land it if you're in a hurry.

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit d9d545ed:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=d9d545ed6c4fd87c38153c4798d2837ecd8dd290).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4098293) pytorch_ios_11_2_1_x86_64_build (1/1)
**Step:** "Run Simulator Tests" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=79269000) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=79269000))
<details>
<summary>
<code>ImportError: cannot import name 'PILLOW_VERSION' from 'PIL' (/Users/distiller/anaconda/lib/python3.7/site-packages/PIL/__init__.py)</code>
</summary>

```
  File "/Users/distiller/anaconda/lib/python3.7/site-packages/torchvision/datasets/__init__.py", line 9, in <module> 
    from .fakedata import FakeData 
  File "/Users/distiller/anaconda/lib/python3.7/site-packages/torchvision/datasets/fakedata.py", line 3, in <module> 
    from .. import transforms 
  File "/Users/distiller/anaconda/lib/python3.7/site-packages/torchvision/transforms/__init__.py", line 1, in <module> 
    from .transforms import * 
  File "/Users/distiller/anaconda/lib/python3.7/site-packages/torchvision/transforms/transforms.py", line 17, in <module> 
    from . import functional as F 
  File "/Users/distiller/anaconda/lib/python3.7/site-packages/torchvision/transforms/functional.py", line 5, in <module> 
    from PIL import Image, ImageOps, ImageEnhance, PILLOW_VERSION 
ImportError: cannot import name 'PILLOW_VERSION' from 'PIL' (/Users/distiller/anaconda/lib/python3.7/site-packages/PIL/__init__.py) 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Pull request closed without merging**